src/databricks/labs/remorph/reconcile/connectors/databricks.py

```diff
@@ -54,9 +56,12 @@ def read_data(
         query: str,
         options: JdbcReaderOptions | None,
     ) -> DataFrame:
-        if not catalog:
-            catalog = "hive_metastore"
-        table_with_namespace = f"{catalog}.{schema}.{table}"
+        namespace_catalog = "hive_metastore" if not catalog else catalog
```

Comment:
add the relevant changes in `core` and a functional test to make we keep this feature in proper implementation.